modules/blockbuilder/blockbuilder.go

```diff
@@ -472,14 +477,20 @@ func (b *BlockBuilder) pushTraces(ts time.Time, tenantBytes, reqBytes []byte, p
 	return p.pushBytes(ts, string(tenantBytes), req)
 }
 
-func (b *BlockBuilder) getAssignedActivePartitions() []int32 {
-	activePartitionsCount := b.partitionRing.PartitionRing().ActivePartitionsCount()
-	assignedActivePartitions := make([]int32, 0, activePartitionsCount)
+// Gets assigned partitions, these can be active or inactive. Pending partitions won't be included
+func (b *BlockBuilder) getAssignedPartitions() []int32 {
+	partitions := b.partitionRing.PartitionRing().Partitions()
+	ringAssignedPartitions := make(map[int32]bool, len(partitions))
+	for _, p := range partitions {
+		if p.IsActive() || p.IsInactive() {
+			ringAssignedPartitions[p.Id] = true
+		}
+	}
+	assignedActivePartitions := make([]int32, 0, len(ringAssignedPartitions))
```

Comment:
Nit: not sure this is worth it. The final length will be `1/N`, with `N` the number of generators